File: waldiez/models/agents/user_proxy/user_proxy.py
# SPDX-License-Identifier: Apache-2.0.
# Copyright (c) 2024 - 2025 Waldiez and contributors.
"""User proxy agent model."""

from pydantic import Field
from typing_extensions import Annotated, Literal

from ..agent import WaldiezAgent
from .user_proxy_data import WaldiezUserProxyData


class WaldiezUserProxy(WaldiezAgent):
    """User proxy agent model.

    A `WaldiezAgent` with agent_type `user` and
    default `human_input_mode`: `"ALWAYS"`
    See `WaldiezAgent`,`WaldiezUserProxyData`,`WaldiezAgentData` for more info.

    Attributes
    ----------
    agent_type : Literal["user", "user_proxy"]
        The agent type: 'user' for a user proxy agent
    data : WaldiezUserProxyData
        The user proxy agent's data
    """

    agent_type: Annotated[
        Literal["user", "user_proxy"],
        Field(
            "user_proxy",
            title="Agent type",
            description=(
                "The agent type in a graph. "
                "`user` is deprecated and will be removed in future versions. "
                "Use `user_proxy` instead."
            ),
            alias="agentType",
        ),
    ]
    data: Annotated[
        WaldiezUserProxyData,
        Field(
            title="Data",
            description="The user proxy agent's data",
            default_factory=WaldiezUserProxyData,
        ),
    ]

File: src/waldiez/models/Agent/UserProxy/UserProxy.ts
/**
 * SPDX-License-Identifier: Apache-2.0
 * Copyright 2024 - 2026 Waldiez & contributors
 */
import { WaldiezAgent, type WaldiezNodeAgentType } from "@waldiez/models/Agent/Common";
import { WaldiezAgentUserProxyData } from "@waldiez/models/Agent/UserProxy/UserProxyData";

/**
 * Waldiez User Proxy Agent.
 * @param id - The id of the user proxy
 * @param type - The type of the node in a graph (agent)
 * @param agentType - The type of the agent (user)
 * @param name - The name of the agent
 * @param description - The description of the agent
 * @param tags - The tags of the agent
 * @param requirements - The requirements of the agent
 * @param createdAt - The creation date of the agent
 * @param updatedAt - The update date of the agent
 * @param data - The data of the agent. See {@link WaldiezAgentUserProxyData}
 */
export class WaldiezAgentUserProxy extends WaldiezAgent {
    data: WaldiezAgentUserProxyData;
    agentType: WaldiezNodeAgentType = "user_proxy";
    constructor(props: {
        id: string;
        name: string;
        description: string;
        tags: string[];
        requirements: string[];
        createdAt: string;
        updatedAt: string;
        agentType: WaldiezNodeAgentType;
        data: WaldiezAgentUserProxyData;
        rest?: { [key: string]: unknown };
    }) {
        super(props);
        this.agentType = "user_proxy";
        this.data = props.data;
        this.rest = props.rest || {};
    }
}
